dvc/tree/_metadata.py

```diff
@@ -0,0 +1,55 @@
+from dataclasses import dataclass, field
```

Comment:
Why `_metadata.py` though?

Reply:
I didn't want to expose it outside of the `tree`s. But, don't have a strong opinion.

Reply:
Good point. We don't use that technique anywhere else for files in the module, but this is indeed a very suitable place to start. Let's keep it.